heron/stmgr/src/cpp/manager/checkpoint-gateway.cpp

```diff
@@ -73,30 +72,34 @@ void CheckpointGateway::SendToInstance(sp_int32 _task_id,
   size_metric_->SetValue(current_size_);
 }
 
-void CheckpointGateway::SendToInstance(proto::stmgr::TupleStreamMessage* _message) {
+void CheckpointGateway::SendToInstance(unique_ptr<proto::stmgr::TupleStreamMessage> _message) {
   if (current_size_ > drain_threshold_) {
     ForceDrain();
   }
+
   sp_int32 task_id = _message->task_id();
   sp_uint64 size = _message->set().size();
-  CheckpointInfo* info = get_info(task_id);
-  _message = info->SendToInstance(_message, size);
-  if (!_message) {
+  CheckpointInfo& info = get_info(task_id);
+
+  proto::stmgr::TupleStreamMessage *raw_message = info.SendToInstance(_message.release(), size);
```

Comment:
I am wondering about the difference of .release() and std::move()?

Reply:
We can not move here because `tuplestream_drainer_` and all the chain down the road expects just a raw pointer. That's why i converted `uniq_ptr` into a raw pointer using `release`. `release` basically tells that this concrete smart pointer is not responsible for managing this concrete raw pointer any more.

Reply:
discussed offline